pkg/apis/garden/v1beta1/types.go

```diff
@@ -758,6 +758,8 @@ type ShootStatus struct {
 	// UID is a unique identifier for the Shoot cluster to avoid portability between Kubernetes clusters.
 	// It is used to compute unique hashes.
 	UID types.UID `json:"uid"`
+	// KubeProxyMode contains the current proxy mode of the shoot cluster.
+	KubeProxyMode string `json:"kubeProxyMode"`
```

Comment:
> Have you tested the behavior within a Shoot cluster?
> 
> I just read in the cmd doc of Kube-Proxy:
> `&o.CleanupAndExit, "cleanup", o.CleanupAndExit, "If true cleanup iptables and ipvs rules and exit."`
> 
> --> Kube Proxy does only the clean-up but doesn't start afterwards?
> This sounds quite disturbing to me since we keep the `--cleanup` flag until the next reconciliation.

Kube Proxy restarts after a change occurs in the ds.

Reply:
Sure it does but it starts with `--cleanup` which lets it only do the clean-up and nothing more (left in a no-op loop), see https://github.com/vllry/kubernetes/blob/3c9989258a695fd9cbd543edb0e3d8ed45daafd9/cmd/kube-proxy/app/server.go#L566

I gave it a try and tested it, the cluster was not functioning any more as long as the `--cleanup` was in the ds.

Reply:
The flag needs to be removed after running the cleanup, either a new reconcile has to be triggered directly or the cleanup has to be called in a separate cmd. I will change that.

Reply:
The next challenge I see here is when it comes to timing:

I didn't see but also didn't check explicitly for any indication and since Kube-Proxy runs in a no-op loop with `--cleanup` (pod runs infinitely), how do we know when the clean-up job is done?